### ISSUE TITLE
Fast-RTPS 1.9.0 no longer builds with libasio-dev from Ubuntu 18.04

### ISSUE THREAD

**Issue by nuclearsandwich** [eProsima/Fast-RTPS#619]
The changes in 1.9.0 appear to have increased the required version of ASIO beyond what is available upstream in Ubuntu 18.04. Were the changes intentional and are they necessary? ROS 2 currently uses ASIO from upstream Ubuntu in order to avoid conflicts between different vendored versions of ASIO so we would prefer to keep building Fast-RTPS with upstream.

Example build failures: https://ci.ros2.org/job/ci_linux/7573/console

**Comment by richiware** [comment]
#616 should fix it.

**Comment by MiguelCompany** [comment]
This should have been solved by #616 

I have just done the following:

```
sudo apt install libasio-dev   
# reports: libasio-dev is already the newest version (1:1.10.8-1).

mkdir src
wget https://raw.githubusercontent.com/ros2/ros2/foonathan-memory-vendor/ros2.repos
vcs import src < ros2.repos
colcon build --packages-up-to fastrtps --event-handlers console_cohesion+ console_package_list+ --cmake-args -DINSTALL_EXAMPLES=OFF -DSECURITY=ON
```

And build has been successful

**Comment by nuclearsandwich** [comment]
Thanks for the quick fix! Is it possible to add CI coverage for using dependencies from upstream when they're available in order to catch issues like this before release?
